### PR TITLE
Add `safe` parameter to `base64.encodeMime`

### PR DESCRIPTION
**Pull request by AmjadHD**
* `string` to `openArray[char or byte]`
* `safe` parameter

### CODE REVIEW

**Comment by AmjadHD** [comment]
https://github.com/nim-lang/Nim/blob/fa60378a7f992235efd766d656556a3e1e337008/lib/pure/base64.nim#L179
`Error: undeclared identifier 'b'`

**Comment by AmjadHD** [comment]
cc @Varriount (I had to ping someone :) )

File: lib/pure/base64.nim [hunk 1]
```diff
@@ -161,10 +161,12 @@ proc encode*[T: byte|char](s: openArray[T], safe = false): string =
...
   encodeImpl()
 
-proc encode*[T: SomeInteger and not byte](s: openArray[T], safe = false): string {.deprecated: "use `byte` or `char` instead".} =
+proc encode*[T: SomeInteger and not byte](s: openArray[T], safe = false): string
+  {.deprecated: "use `byte` or `char` instead".} =
   encodeImpl()
 
-proc encodeMime*(s: string, lineLen = 75.Positive, newLine = "\r\n"): string =
+proc encodeMime*[T: char|byte](s: openArray[T], lineLen = 75.Positive,
+                               newLine = "\r\n", safe = false): string =
```
Comment (Varriount) [inline lib/pure/base64.nim]: You need to add documentation on what `safe` does.

**Comment by Araq** [comment]
Why can `mime` be anything other but string? Makes no sense to me.

**Comment by AmjadHD** [comment]
I don't know.

**Pull request merged**